### PR TITLE
Document packaging and versioning

### PR DESCRIPTION
**Pull request by chaen**
closes https://github.com/DIRACGrid/diracx/issues/145

### CODE REVIEW

File: docs/VERSIONING.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+     └────────────────────────┘              └─────────────────────┘
+
+```
+
+### Dependencies
+
+- There is a noted duplication between `setup.cfg` and `environment.yaml`.
+- The `diracx/base` image is built from a Dockerfile with `environment.yml`, primarily defining the Python version.
+- The `diracx/services-base` and `diracx/tasks-base` images extend `diracx/base` with additional Dockerfiles and `environment.yml`, tailored to their specific needs.
+- The `diracx/services` and `diracx/tasks` images are further built upon their respective base images, adding necessary diracx packages through `pip install --no-dependencies`.
```
Comment (fstagni) [inline docs/VERSIONING.md:91]: is "images" the right name here?

File: docs/VERSIONING.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+## Extensions
+
+- Extensions one or more of `diracx`, `diracx-routers`, `diracx-tasks` (e.g. `mydiracx`, `mydiracx-routers`, `mydiracx-tasks`).
```
Comment (fstagni) [inline docs/VERSIONING.md:95]: Not sure what you are saying here.

File: docs/VERSIONING.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+└─────────┘   │   │        └──┬──┘
+            ┌─▼───▼─┐      ┌──▼──┐
+            │ tasks │      │ cli │
+            └───────┘      └─────┘
+```
+
+
+### Versioning Strategy
+
+- Currently, we employ a versioning format of v0.<major>.<minor>.
```
[the remark this answers is not in the document]
Comment (chaen) [reply docs/VERSIONING.md:47]: We don't know exactly how we will want to manage release later, but as long as we still live with both `DIRAC` and `diracx`, we will stick with `v0`

File: docs/VERSIONING.md (hunk 1 above)
Comment (chaen) [reply docs/VERSIONING.md:91]: I think so. What else do you have in mind ?

File: docs/VERSIONING.md (hunk 2 above)
Comment (chaen) [reply docs/VERSIONING.md:95]: An extension that would have only services and no tasks would create `mydiracx-routers`, but not `mydiracx-tasks`

File: docs/VERSIONING.md (hunk 1 above)
Comment (fstagni) [reply docs/VERSIONING.md:91]: It's probably me getting confused: are `diracx/services` and `diracx/tasks` docker images?

File: docs/VERSIONING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (fstagni) [reply docs/VERSIONING.md:47]: I'd add your comment to the above.

[the remark this answers is not in the document]
Comment (chrisburr) [reply docs/VERSIONING.md:47]: Maybe `v0.<major>.<patch>` better communicates what we have in mind for now.

File: docs/VERSIONING.md (hunk 1 above)
Comment (chrisburr) [reply docs/VERSIONING.md:91]: Yes

File: docs/VERSIONING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (chaen) [reply docs/VERSIONING.md:47]: OK, added

File: docs/VERSIONING.md (hunk 2 above)
Comment (chaen) [reply docs/VERSIONING.md:95]: done

**Review by fstagni — APPROVED** [review]

**Pull request merged**